docs/api-reference/map.md

```diff
@@ -26,20 +26,21 @@ function App() {
 Imperative methods are accessible via a [React ref](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) or the [useMap](/docs/api-reference/use-map.md) hook.
 
 
-```js
+```tsx
 import * as React from 'react';
 import Map from 'react-map-gl';
+import mapboxgl from 'mapbox-gl';
 
 function App() {
-  const mapRef = React.useRef();
+  const mapRef = React.useRef<MapRef<mapboxgl.Map>>();
 
   const onMapLoad = React.useCallback(() => {
     mapRef.current.on('move', () => {
       // do something
     });
   }, []);
 
-  return <Map mapLib={import('mapbox-gl')} ref={mapRef} onLoad={onMapLoad} />;
+  return <Map mapLib={mapboxgl} ref={mapRef} onLoad={onMapLoad} />;

```

Comment:
It's unclear to me what kind of type hinting `<Map/>` provides now.. will it include `mapboxgl` parameters?

Reply:
Map props will be typed following the options expected by the `mapLib.Map` constructor.